lib/compress/zstd_compress.c

```diff
@@ -6918,11 +6980,13 @@ ZSTD_compressSequences_internal(ZSTD_CCtx* cctx,
         }
 
         RETURN_ERROR_IF(dstCapacity < ZSTD_blockHeaderSize, dstSize_tooSmall, "not enough dstCapacity to write a new compressed block");
-        compressedSeqsSize = ZSTD_entropyCompressSeqStore(&cctx->seqStore,
-                                &cctx->blockState.prevCBlock->entropy, &cctx->blockState.nextCBlock->entropy,
-                                &cctx->appliedParams,
+        compressedSeqsSize = ZSTD_entropyCompressSeqStore_wExtLitBuffer(
```

Comment:
It looks like this is the same as `ZSTD_entropyCompressSeqStore()`, can we call that instead? Or am I missing something?

Reply:
`ZSTD_entropyCompressSeqStore()` is invoking `ZSTD_entropyCompressSeqStore_wExtLitBuffer()`,
so they both converge to the same implementation.

The difference is that `ZSTD_entropyCompressSeqStore()` forces the `literals` buffer to be the one reserved into `cctx`, while `_wExtLitBuffer()` let the caller select where the buffer of `literals` is.

Reply:
@Cyan4973 I understand that, but in this case the literals are in the seqStore, right?

There are so many parameters that it is hard to read, but it looks like this could just invoke `ZSTD_entropyCompressSeqStore()` instead, because the literals are `cctx->seqStore.litStart, (size_t)(cctx->seqStore.lit - cctx->seqStore.litStart)`.

Reply:
Ah, you mean this specific instance.
And yes, you are right, it could be replaced by `ZSTD_entropyCompressSeqStore()`.

This modification was made early during development, when I was trying to have a common pipeline for the 2 variants. This is a left over from this period. Since I gave up this approach, we now have 2 separate pipelines, so it's possible to restore the call to `ZSTD_entropyCompressSeqStore()`, especially if it helps readability.